toptal/picasso
# feat(PageFooter): [SCR-2645] make Footer copyright customizable

[SCR-2645]

### Description

Allows to customize Page Footer copyright for the services created by Toptal, but should be branded as separate 👇 
<img width="372" alt="Screenshot 2022-01-31 at 10 17 52" src="https://user-images.githubusercontent.com/5591973/151763183-64bbf24e-d05c-451f-a43a-43a0b5b6cdda.png">

### Tech Description
Footer lives in several examples thus I had to update too many snapshots.


### How to test

- Visit [page component](https://picasso.toptal.net/scr-2645-page-footer-add-custom-logo/?path=/story/layout-page--page)
- Click on `Page.Footer`
- **Action**: Check `ContentCopyright` property
- <img width="860" alt="Screenshot 2022-01-31 at 11 01 18" src="https://user-images.githubusercontent.com/5591973/151765557-164cb252-7ecd-4ffb-971b-f37fc3a28254.png">
- Open [example](https://picasso.toptal.net/scr-2645-page-footer-add-custom-logo/?path=/story/layout-page--page#copyright-content)
- **Action**: Check how you can override content
- <img width="1158" alt="Screenshot 2022-01-31 at 11 02 03" src="https://user-images.githubusercontent.com/5591973/151765649-7b1f7684-5e4e-428c-9774-fd19e5d3747f.png">


### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| <img width="1154" alt="Screenshot 2022-01-31 at 10 57 36" src="https://user-images.githubusercontent.com/5591973/151765125-f0c0fe4a-df67-4355-a9d3-be72568f452d.png"> | <img width="1154" alt="Screenshot 2022-01-31 at 10 57 36" src="https://user-images.githubusercontent.com/5591973/151765100-ba318b3c-7dd5-48c5-88cb-7c1b0f490f43.png"> | 

### Review


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[SCR-2645]: https://toptal-core.atlassian.net/browse/SCR-2645?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
